### ISSUE TITLE
SDP Constraints: Constrain the Magnitude of Eigenvalues of a Weighting Matrix

### ISSUE THREAD

**Issue by SamChevalier** [Lezcano/geotorch#18]
Hello, thank you for the awesome Geotorch tools! I am wondering if it is possible to use Geotorch to constrain the magnitude of the eigenvalues of a (symmetric) weighting matrix W in a Neural Network? These constraints would take the following form (if I want to constrain them to be smaller than 1 in magnitude):

<img width="106" alt="Eig_Const" src="https://user-images.githubusercontent.com/39089645/115120157-3c500300-9fac-11eb-92b1-de70be51b5dc.PNG">

If so, how could these constraints be posed in the code? Thank you!

**Comment by Lezcano** [comment]
This is indeed possible, although it is not exactly documented :)

Having a look at 
https://geotorch.readthedocs.io/en/latest/psd/psd.html
we see that `PSD` takes a function that is used to map the eigenvalues from R to some subset. We use a softmax by default, but we could use any function for this. In your case, you want the eigenvalues to be in the interval `(-1, 1)`. As such, you could pass the function `torch.tanh` or `torch.sin` or a rescaled `torch.sigmoid`.

If you want to do this using the function [geotorch.positive_definite](https://geotorch.readthedocs.io/en/latest/constraints.html#geotorch.positive_definite), you should pass that function and its inverse in a tuple. For example:
```python
linear = nn.Linear(8,8)
geotorch.positive_definite(linear, f=(torch.tanh, torch.atanh))
```
You can see examples of these pairs of functions and their inverses that map R -> (-1, 1) in this other code of GeoTorch:
https://github.com/Lezcano/geotorch/blob/553a38f78402b01dc9f15dfa4109a46afa7463a9/geotorch/almostorthogonal.py#L7-L21

**Comment by SamChevalier** [comment]
This is very helpful - thank you for the quick reply! This indeed does work nicely for constraining the eigenvalues to be less than 1 in magnitude.

**Comment by Lezcano** [comment]
I'll close this for now, feel free to reopen if you find any problem :)
